### DISCUSSION TITLE
Feature Suggestion: Automated Threat Mapping and Control Correlation

### DISCUSSION THREAD

**Discussion by z4l1nux** [OWASP/threat-dragon#992] [Ideas]
**Description:**
I propose adding a feature to OWASP Threat Dragon that automatically maps threats and mitigations from STRIDE to CAPEC and CWE, and allows for the search of attacks and controls within the tool's interface. This enhancement would streamline the threat modeling process and provide users with direct access to detailed information about threats and mitigations.

**Key Features:**
Automated Threat Mapping:

- After drawing a DFD and identifying threats using STRIDE, the tool will automatically assign related threats and mitigations from CAPEC and CWE.

**Search Functionality:**

- Users will have the ability to search for specific attacks and controls in CAPEC and CWE directly from the Threat Dragon interface.

**Correlation with ASVS:**

- Map DFD STRIDE threats to CAPEC -> CWE -> ASVS controls, providing a comprehensive view of mitigations and compliance.

**Report Generation:**

- Generate detailed reports that include DFD, identified threats, suggested mitigations, and correlations with ASVS controls.

**Benefits:**

- Streamlines the threat modeling process.

- Provides easy access to comprehensive threat and mitigation data.

- Enhances the value of Threat Dragon by integrating with established security standards (CAPEC, CWE, ASVS).

**Comment by jgadsden** [comment]
certainly it sounds attractive, although it may be a lot of work initially

**Reply by z4l1nux** [reply]
Hello Threat Dragon community!
I wanted to share some ideas I've been working on and get your feedback. I'm exploring a way to enhance Threat Dragon by integrating AI capabilities to automatically analyze component properties and descriptions (Processes, Actors, etc.) and map them to appropriate threats using the STRIDE and CAPEC frameworks.

**The basic concept:**

1. Use AI provider APIs (like OpenAI, Claude, etc.) to analyze the properties and descriptions of diagram components
2. Automatically map these components to relevant STRIDE categories
3. Associate each STRIDE category with corresponding CAPEC entries based on the mapping found at https://ostering.com/blog/2022/03/07/capec-stride-mapping/
4. Automatically apply appropriate threats to the components

I've already made some progress on implementing this concept and would like to share some screenshots of what I've done so far. Please excuse any errors and the fact that some components are still labeled in Portuguese - I'm in the process of translating everything.

**Questions for the community:**

1. Do you think this AI-assisted threat mapping would be valuable for the Threat Dragon project?
2. Have others tried similar approaches to automating the threat modeling process?
3. Are there particular challenges or limitations you foresee with this approach?
4. What additional features would make this more useful for threat modeling teams?
5. Would this be something the community would be interested in contributing to or using?

I believe this approach could help teams more quickly identify potential threats and streamline the threat modeling process, especially for those who are newer to security concepts. However, I want to ensure this adds genuine value to the project rather than unnecessary complexity.
Looking forward to your thoughts and feedback!

![Captura de tela 2025-03-14 121247](https://github.com/user-attachments/assets/b737f553-a5cd-42c2-adcd-fbf005b0ec2f)
![Captura de tela 2025-03-14 120135](https://github.com/user-attachments/assets/f0b11d34-3fb8-45e9-a60b-588ade56fc81)
![Captura de tela 2025-03-14 121341](https://github.com/user-attachments/assets/ab15401c-21b6-490e-bca4-0bd73ee84bad)
![Captura de tela 2025-03-14 121403](https://github.com/user-attachments/assets/850053de-a079-47cb-9dca-b8a2dbcea585)
![Captura de tela 2025-03-14 120304](https://github.com/user-attachments/assets/766f6635-6680-425b-bed9-d65690b73ec5)
